Samba 4.9.13

DavidePrincipi opened the pull request:
NOTE: do not merge-squash!

https://github.com/NethServer/dev/issues/5838

DavidePrincipi committed:
Samba 4.9.13

markVnl committed:
ARM support for nethserver-dc builds

- Make Spec arch indifferent
- Also do not package empty debug

DavidePrincipi committed:
Fix install/upgrade actions for ARM

markVnl commented on nethserver-dc.spec:
```diff
@@ -6,7 +16,7 @@ Summary:        NethServer Domain Controller configuration
 Source0:        %{name}-%{version}.tar.gz
-Source1:        https://github.com/NethServer/ns-samba/releases/download/4.9.13/ns-samba-4.9.13-1.ns7.x86_64.rpm
+Source1:        https://github.com/NethServer/ns-samba/releases/download/4.9.13/ns-samba-4.9.13-1.ns7.%{source1_arch}.rpm
 
```
Nothing really somthing wrong but  we have to keep in mind the source-package will differ for different arch's.
IOW you can't rebuild a source package produced on x86_64 on arm. 

IMHO it is good enough for us, for reference the solution how it "should be" :

Download all ns-samba arch sources so they are included in the SPRM.
```
Source1:        https://github.com/NethServer/ns-samba/releases/download/4.8.12/ns-samba-4.8.12-1.ns7.x86_64.rpm
Source2:        https://github.com/NethServer/ns-samba/releases/download/4.8.12/ns-samba-4.8.12-1.ns7.aarch64.rpm
Source3:        https://github.com/NethServer/ns-samba/releases/download/4.8.12/ns-samba-4.8.12-1.ns7.arm7hl.rpm
```


then @ `%install ` do something like this:

```
%ifarch %{arm}
mv %{SOURCE3} root/usr/lib/nethserver-dc
%elif
%ifarch %{aarch64}
mv %{SOURCE2} root/usr/lib/nethserver-dc
%else
mv %{SOURCE1} root/usr/lib/nethserver-dc
%endif
```

But this means arm variants of ns-samba **must**  be available during x86_64 builds.
IMO we cannot take that risk..

DavidePrincipi replied:
The RPM within another RPM **is the real issue**. The correct approach IMO is adding another YUM repository for the `ns-samba` package, enabled only during `nsdc` install/upgrades.

markVnl replied:
yes , but we want to be sure it is never installed outside the container. 
that is the reason it is in there isn't it?

DavidePrincipi replied:
yes, because it conflicts with upstream `samba` packages. They can't be in the same repo, but I hope nsdc installation can work with an extra repository.

markVnl commented on nethserver-dc.spec:
```diff
@@ -1,3 +1,13 @@
+# Naming of 32bit arm is a mess..
+%ifarch %{arm}
+%define source1_arch %{__isa_name}
+%else
```
Have second thoughts about the expansion of `%{__isa_name}` after reading the evebox threat on the community forum. And decided to hard-code the arm arch

`%define source1_arch armv7hl`

DavidePrincipi replied:
We can go with it, until we decide `aarch64` is interesting too!

markVnl replied:
On aarch64 `%define source1_arch %{_arch}` expands just fine to `aarch64` , So we are good to go here too

DavidePrincipi replied:
Please tell me if I understand correctly. Should be changed to:

```spec
# Naming of 32bit arm is a mess..
%ifarch armv7hl
%define source1_arch armv7hl
%else
%define source1_arch %{_arch}
%endif
```

markVnl replied:
yes, this work's for all arch's

markVnl committed:
Make an exception for armv7hl only

Have second thoughts about the expansion of %{__isa_name}
after reading the evebox thread on the community forum.
